zkvm/src/predicate.rs

```diff
@@ -136,42 +118,35 @@ impl Predicate {
         Predicate::Opaque(self.to_point())
     }
 
-    /// Creates a disjunction of a vector of predicates.
-    pub fn disjunction(preds: Vec<Predicate>) -> Result<Self, VMError> {
-        let point = {
-            let f = Predicate::commit_disjunction(preds.iter().map(|p| p.to_point()));
-            let l = preds[0]
-                .to_point()
-                .decompress()
-                .ok_or(VMError::InvalidPoint)?;
-            l + f * PedersenGens::default().B
-        };
-        Ok(Predicate::Or(PredicateDisjunction {
-            preds: preds,
-            precomputed_point: point.compress(),
-        }))
-    }
-
-    fn commit_disjunction<I>(preds: I) -> Scalar
-    where
-        I: IntoIterator,
-        I::Item: Borrow<CompressedRistretto>,
-        I::IntoIter: ExactSizeIterator,
-    {
-        let mut t = Transcript::new(b"ZkVM.predicate");
-        let iter = preds.into_iter();
-        t.commit_u64(b"n", iter.len() as u64);
-        for x in iter {
-            t.commit_point(b"X", x.borrow());
+    fn commit_taproot(key: &[u8], root: &[u8]) -> Scalar {
+        let mut t = Transcript::new(b"ZkVM.taproot");
+        t.commit_bytes(b"key", key);
+        t.commit_bytes(b"merkle", root);
+        t.challenge_scalar(b"h")
+    }
+
+    /// Verifies whether the current predicate is a commitment to a signing key `key` and Merkle root `root`.
+    /// Returns a `PointOp` instance that can be verified in a batch with other operations.
+    pub fn prove_taproot(&self, program_item: &ProgramItem, call_proof: &CallProof) -> PointOp {
+        let key = &call_proof.verification_key;
+        let neighbors = &call_proof.neighbors;
+        let root = MerkleTree::compute_root_from_path(b"ZkVM.taproot", program_item, neighbors);
+        let h = Self::commit_taproot(&key.into_compressed().to_bytes(), &root);
```

Comment:
every use of `commit_taproot` has to convert a verification key to byteslice. How about we do so inside that function instead and take `key:&VerificationKey` parameter?

```suggestion
        let h = Self::commit_taproot(&key, &root);
```